pytorch_lightning/callbacks/gpu_stats_monitor.py

```diff
@@ -90,78 +104,95 @@ def on_train_start(self, trainer, pl_module):
             )
 
         if not trainer.on_gpu:
-            rank_zero_warn(
-                'You are using GPUStatsMonitor but are not running on GPU.'
-                ' Logged utilization will be independent from your model.', RuntimeWarning
+            raise MisconfigurationException(
+                'You are using GPUStatsMonitor but are not running on GPU'
+                f' since gpus attribute in Trainer is set to {trainer.gpus}.'
             )
 
+        self._gpu_ids = ','.join(map(str, trainer.data_parallel_device_ids))
+
     def on_train_epoch_start(self, trainer, pl_module):
-        self.snap_intra_step_time = None
-        self.snap_inter_step_time = None
+        self._snap_intra_step_time = None
+        self._snap_inter_step_time = None
 
     @rank_zero_only
     def on_train_batch_start(self, trainer, pl_module, batch, batch_idx, dataloader_idx):
-        if self._log_stats.gpu_utilization:
-            self._log_usage(trainer)
+        gpu_stat_keys = []
+        gpu_stat_keys.extend(self._get_gpu_stat_keys())
```

Comment:
you do this to copy the list? but do we really need to? does it get mutated anywhere?

Reply:
no, was just playing around a bit with this, forgot to put it back :)